### ISSUE TITLE
Review Timezone / DateTime Utils

### ISSUE THREAD

**Issue by inveterateliterate** [LaunchPadLab/rails_util#38]
timezone helper may be broken ...

review: Empower, Digital Dental

@dianacamacho has a helper she's been using...

**Comment by dianacamacho** [comment]
@inveterateliterate not sure if the output of the two is the same or not (or the manner in which its used in existing apps), but what I've been using is below (i've had no problems with it so far - that i know of...) 

the dependence on "offsets" is what always threw me off in the current rails util timezone helper and why i went with my own option (and might be why it might be broken? since offsets for a given zone change depending on time of year (daylight savings))

my example isn't relying on any offset calculations, it just takes the current timestamp or DT object you want to convert and builds a new DT object in the target zone with the year/month/day/hour/min/sec of the original DT. 

```rb
 DEFAULT_TIMEZONE = ENV["APP_TIMEZONE"] || "UTC"

  # changes the zone of a given datetime (string or object) to the provided target timezone,
  # ex: passing 10:00:00 UTC datetime with target zone of "America/New_York" 
  # returns a datetime object set to 10:00:00 ET
  def datetime_in_zone(datetime, zone = DEFAULT_TIMEZONE)
    DateTime.new
      .in_time_zone(zone)
      .change(
        year: datetime.to_datetime.year,
        month: datetime.to_datetime.month,
        day: datetime.to_datetime.day,
        hour: datetime.to_datetime.hour,
        min: datetime.to_datetime.min,
        sec: datetime.to_datetime.sec
      )
  end
 ```

i use the following to build a DT object manually in a target zone
```rb
def build_datetimebuild_datetime(date:, hour: 0, min: 0, sec: 0, timezone: DEFAULT_TIMEZONE)
  time_now = date.to_datetime || current_datetime
  datetime = DateTime.new(time_now.year, time_now.month, time_now.day, hour, min, sec)
  datetime_in_zone(datetime, timezone)
end

def current_datetime(zone = DEFAULT_TIMEZONE)
  DateTime.current.in_time_zone(zone)
end
```

**Comment by inveterateliterate** [comment]
yes thank you, I was not clear in my note on this issue - I was more proposing your utils getting added to these collection of utils but also that the existing util needs review.

**Comment by danparnella** [comment]
I'm pretty sure we just ran into this on Jet It as well. Now that we're out of Daylight Savings time the util is trying to convert a datetime without DST even though the provided date is April (so should have DST applied). https://github.com/LaunchPadLab/jet-it-api/pull/103

cc: @heyapricot

**Comment by inveterateliterate** [comment]
lol I just looked up this issue too!
